bears/coffee_script/CoffeeLintBear.py

```diff
@@ -22,11 +23,289 @@ class CoffeeLintBear:
+            Checks if there is number equal ``number_of_lines_after_classes`` of
+            newlines after classes definition.
+        :param number_of_newlines_after_classes:
+            Determines the number of newlines that separate the class definition
+            and the rest of the code.
+        :param no_backticks:
+            Prevents some JavaScript elements like ``eval`` to affect
+            CoffeeScript.
+        :param no_implicit_braces:
+            Prohibits implicit braces when declaring object literals.
+
+            Example: If ``no_implicit_braces = True`` then
+            ```
+            1:2, 3:4
+            ```
+            is prohibited, whereas
+            ```
+            {1:2, 3:4}
+            ```
+            is accepted.
+        :param no_implicit_parentheses:
+            Prohibits implicit parentheses.
+        :param no_interpolation_in_single_quotes:
+            Prohibits string interpolation in a single quoted string.
+
+            Example: If ``no_interpolation_in_single_quotes = True`` then
+            ```
+            f = '#{bar}'
+            ```
+            is prohibited, whereas
+            ```
+            f = "#{bar}"
+            ```
+            is correct.
+        :param no_stand_alone_at:
+            Prohibits the use of stand alone  ``@``.
+
+            Example: If ``no_stand_alone_at = True``
+            ```
+            @ notok
+            not(@).ok
+            @::
+            ```
+            are prohibited, whereas
+            ```
+            @alright
+            @(fn)
+            @ok()
+            @[ok]
+            @ok()
+            ```
+            are accepted.
+        :param disable_throwing_strings:
+            Disables throwing string literals or interpolation. It is forbidden
+            by default.
+
+            Example: If ``disable_throwing_strings = True``
+            ```
+            throw 'my error'
+            throw "#{1234}"
+            ```
+            will not be permitted.
+        :param no_trailing_semicolons:
+            Prohibits trailing semicolons since they are not useful.
+            The semicolon is meaningful as there's another instruction on
+            the same line.
+
+            Example: If ``no_trailing_semicolon = True``
+            ```
+            x = '1234'; console.log(x)
+            ```
+            here the semicolon is meaningful.
+            ```
+            alert('end of line');
+            ```
+            the semicolon is redundant.
+        :param no_trailing_whitespaces:
+            Checks whether they are trailing whitespace in the code or not.
+        :param no_trailing_whitespaces_affects_comments:
+            Determines if ``no_trailing_whitespaces`` rule should be applied to
+            comments.
+        :param no_trailing_whitespaces_affects_empty_lines:
+            Determines if ``no_trailing_whitespace`` rule should be applied to
+            empty lines.
```

Comment:
thanks for the review

2016-07-09 19:58 GMT+01:00 Lasse Schuirmann <notifications@github.com>:

> In bears/coffee_script/CoffeeLintBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/555#discussion_r70170809>
> :
>
> > +            ```
> > +            x = '1234'; console.log(x)
> > +            ```
> > +            here the semicolon is meaningful.
> > +            ```
> > +            alert('end of line');
> > +            ```
> > +            the semicolon is redundant.
> > +        :param no_trailing_whitespaces:
> > +            Checks whether they are trailing whitespace in the code or not.
> > +        :param no_trailing_whitespaces_affects_comments:
> > +            Determines if ``no_trailing_whitespaces`` rule should be applied to
> > +            comments.
> > +        :param no_trailing_whitespaces_affects_empty_lines:
> > +            Determines if ``no_trailing_whitespace`` rule should be applied to
> > +            empty lines.
>
> I wouldn't even expose those two options, is there any real usecase? Just
> set them to true always?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/555/files/19476cff771d8f9515277837c2464aca6b77b493#r70170809>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AJ9xoqbTN0jg7PpRFO_96UyodHhg4Z_xks5qT-9ngaJpZM4I9f6t>
> .
>